src/index.js

```diff
@@ -35,13 +30,10 @@ export default function supernova(env) {
       const constraints = useConstraints();
       const selectionsAPI = useSelections();
       const theme = useTheme();
-
       const [pageInfo, setPageInfo] = useState({ top: 0, height: 100 });
-      const [muiParameters] = useState(muiSetup(__OPIONAL_THEME_DEPS__));
-      const [tableData] = usePromise(() => manageData(model, layout, pageInfo), [layout, pageInfo]);
 
       useEffect(() => {
-        if (layout && tableData) {
+        manageData(model, layout, pageInfo).then((tableData) => {
```

Comment:
Doesn't this mean we run `manageData` everytime `isModal` or `constraints` changes? We only want tor run that if the data has changed.

Reply:
I found that every time isModal or constraints changes, component() runs. so manageData need to run anyway.

Reply:
Same thing here, `tableData` should only update when the data is changed, and then stored. I thought that was what `usePromise` did.

Reply:
You are right. Gonna revert it.